docs/mentoring.md

```diff
@@ -0,0 +1,44 @@
+# Mentoring
+
+To help you get up to speed, or further develop you Argo Workflows contribution skills, you can now get a mentor.
+
+Mentors will help you with things like:
+
+* Resources to learn Golang, Typescript and React.
+* Get Argo Workflows running in development mode on your laptop.
+* Understand key concepts and learn your way around the source code
+* Getting your first pull request with a code contribution created, reviewed and merged.
+
+Mentors and mentees must abide by the [code of conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md). 
+
+## How To Get A Mentor
+
+Create a pull request which changes this page to add you to this list:
+
+Mentees:
+
+* Add your name here!
+
+We'll find the right mentor for you, and let you know who to add to your PR. Then, your mentor will approve and merge your PR!
+
+## How To Become A Mentor
+
+Create a pull request which changes this page to add you to this list:
+
+Mentees:
+
+* @alexec (Alex)
+* @simster7 (Simon)
+* @sarabala1979 (Bala)
+* @terrytangyuan (Yuan)
+
+## How To End Your Relationship
```

Comment:
Should we make it explicit that the relationship is supposed to end after the mentee's first PR is merged?

Reply:
I think it can go longer.

Reply:
makes sense, we can always adjust later if it turns into a bottleneck.